crates/wasi-http/src/p3/body.rs

```diff
@@ -536,4 +537,12 @@ where
         self.close(res);
         Poll::Ready(Ok(StreamResult::Dropped))
     }
+
+    fn try_into(me: Pin<Box<Self>>, ty: TypeId) -> Result<Box<dyn Any>, Pin<Box<Self>>> {
+        if ty == TypeId::of::<Self>() {
+            Ok(Box::new(*Pin::into_inner(me)))
```

Comment:
This `Box::new(*...)` shouldn't be necessary I think? In theory `Pin::into_inner(me)` should suffice here I believe

Reply:
The compiler doesn't like that:
```
error[E0277]: `dyn std::any::Any` cannot be unpinned
   --> crates/wasi-http/src/p3/body.rs:543:32
    |
543 |             Ok(Pin::into_inner(me))
    |                --------------- ^^ the trait `std::marker::Unpin` is not implemented for `dyn std::any::Any`
    |                |
    |                required by a bound introduced by this call
```